Multiple `pytest` tests fail and testing hangs

### Description

When I run `pytest` without the `-x` option (i.e. when I add the following* to `docker-compose.test.yml` and then run `$ make up-test` followed by `$ make test`), **multiple tests fail... and testing hangs**/freezes.

*here's what I add to `docker-compose.test.yml` so that `$ make test` causes `pytest` to run without the `-x` option (this overrides the `ENTRYPOINT` written in `test.Dockerfile`, which includes the `-x` option):

```diff
  test:
    build:
      context: .
      dockerfile: nmdc_runtime/test.Dockerfile
    container_name: test
+   entrypoint:
+     - "./wait-for-it.sh"
+     - "fastapi:8000"
+     - "--strict"
+     - "--timeout=300"
+     - "--"
+     - "pytest"
    env_file:
```

### Screenshot

Here's a screenshot of the output of `$ make test`, with the above modification to `docker-compose.test.yml` in place. This shows that multiple tests have failed. Also, testing has hung/froze at the point shown (in the sense that, after waiting for more than 10 minutes—specifically timed—there was no change in the output).

![image](https://github.com/microbiomedata/nmdc-runtime/assets/134325062/88e19f45-6a7a-4b63-b68c-02dfb3f8ff7d)

### Environment:
- Docker host: Apple M1 Pro running macOS Ventura 13.4.1 (c)
- Docker Desktop: 4.21.1 (114176) 
- Docker Engine: 24.0.2
- Docker Compose: v2.19.1
- Git commit f812f7bc303ba840c93b4d7dd7dfef4665e1e3d8 (plus the change to `docker-compose.test.yml` shown above)

I pressed `^C` (once or twice) to terminate the testing. 

![image](https://github.com/microbiomedata/nmdc-runtime/assets/134325062/f3f4e1d3-3a60-491b-aee9-708b19f16d74)

Once I did that, the `==== FAILURES ====` section appeared. It showed information about each of the 3 failures that had occurred up to that point. I will post about each one below.

### 1. test_nmdc_jsonschema_using_new_id_scheme

```py
________________________________________________________________________________________________________________________ test_nmdc_jsonschema_using_new_id_scheme ________________________________________________________________________________________________________________________

    def test_nmdc_jsonschema_using_new_id_scheme():
        # nmdc_database_collection_instance_class_names
        for class_name, defn in get_nmdc_jsonschema_dict()["$defs"].items():
            if "properties" in defn and "id" in defn["properties"]:
                if "pattern" in defn["properties"]["id"]:
                    if not defn["properties"]["id"]["pattern"].startswith("^(nmdc):"):
>                       pytest.fail(f"{class_name}.id: {defn['properties']['id']}")
E                       Failed: ChemicalEntity.id: {'description': 'A unique identifier for a thing. Must be either a CURIE shorthand for a URI or a complete URI', 'pattern': '^[a-zA-Z0-9][a-zA-Z0-9_\\.]+:[a-zA-Z0-9_][a-zA-Z0-9_\\-\\/\\.,]*$', 'type': 'string'}

tests/test_util.py:43: Failed
```

### 2. test_nmdc_jsonschema_validator

```py
_____________________________________________________________________________________________________________________________ test_nmdc_jsonschema_validator _____________________________________________________________________________________________________________________________

    def test_nmdc_jsonschema_validator():
        with open(REPO_ROOT.joinpath("metadata-translation/examples/study_test.json")) as f:
            study_test = json.load(f)
            try:
>               _ = nmdc_jsonschema_validator(study_test)

tests/test_util.py:50: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
<string>:298: in validate_https___w3id_org_nmdc_nmdc
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

data = {'description': 'A metagenomic study to couple microbial communities to carbon and contaminant biogeochemistry in the ...e).', 'doi': {'has_raw_value': '10.25585/1487765'}, 'ecosystem': 'Environmental', 'ecosystem_category': 'Aquatic', ...}, custom_formats = {}
name_prefix = 'data.study_set[2]'

>   ???
E   fastjsonschema.exceptions.JsonSchemaValueException: data.study_set[2] must not contain {'doi'} properties

<string>:658: JsonSchemaValueException

During handling of the above exception, another exception occurred:

    def test_nmdc_jsonschema_validator():
        with open(REPO_ROOT.joinpath("metadata-translation/examples/study_test.json")) as f:
            study_test = json.load(f)
            try:
                _ = nmdc_jsonschema_validator(study_test)
            except JsonSchemaValueException as e:
>               pytest.fail(str(e))
E               Failed: data.study_set[2] must not contain {'doi'} properties

tests/test_util.py:52: Failed
```

### 3. test_minter_api_mint

```py
__________________________________________________________________________________________________________________________________ test_minter_api_mint __________________________________________________________________________________________________________________________________

    def test_minter_api_mint():
        client = _get_client()
>       rv = client.request(
            "POST", "/pids/mint", {"schema_class": schema_class, "how_many": 1}
        ).json()

tests/e2e/test_minter_api.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
nmdc_runtime/site/resources.py:60: in request
    self.ensure_token()
nmdc_runtime/site/resources.py:43: in ensure_token
    self.get_token()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <nmdc_runtime.site.resources.RuntimeApiSiteClient object at 0xffff7b0dfee0>

    def get_token(self):
        rv = requests.post(self.base_url + "/token", data=self.get_token_request_body())
        self.token_response = rv.json()
        if "access_token" not in self.token_response:
>           raise Exception(f"Getting token failed: {self.token_response}")
E           Exception: Getting token failed: {'detail': 'Incorrect client_id or client_secret'}

nmdc_runtime/site/resources.py:52: Exception
```

I get an error when I simply `make test`:

```
failed to receive status: rpc error: code = Unavailable desc = error reading from server: EOF
make: *** [test-build] Error 17
```

--- a/.github/pull_request_template.md
+++ b/.github/pull_request_template.md
@@ -1,0 +1,36 @@
+# Description
+
+Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Replace `# (issue)` below with a reference to the issue that this PR addresses, of the form `#\d+`. If merging this PR should close the issue, use `Fixes` or `Closes`. Then delete this paragraph.
+
+Fixes # (issue)
+
+## Type of change
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration, if it is not simply `make up-test && make test-run`.
+
+- [ ] Test A
+- [ ] Test B
+
+**Configuration Details**: none
+
+# Checklist:
+
+- [ ] My code follows the style guidelines of this project (have you run `black nmdc_runtime/`?)
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation (in `docs/` and in <https://github.com/microbiomedata/NMDC_documentation/>?)
+- [ ] I have added tests that prove my fix is effective or that my feature works, incl. considering downstream usage (e.g. <https://github.com/microbiomedata/notebook_hackathons>) if applicable.
+- [ ] New and existing unit and functional tests pass locally with my changes (`make up-test && make test-run`)
+
+

